pkg/operators/extractors/compile.go

```diff
@@ -4,12 +4,23 @@ import (
 	"fmt"
 	"regexp"
 	"strings"
+	"sync"
+	"sync/atomic"
 
 	"github.com/Knetic/govaluate"
 	"github.com/itchyny/gojq"
 	"github.com/projectdiscovery/nuclei/v3/pkg/operators/common/dsl"
 )
 
+var (
+	extractorRegexCache        sync.Map // map[string]*regexp.Regexp
```

Comment:
instead of a sync.Map, we can use a expirable-lru or something with a `TTL` or max-count set so we can automatically do evictions / not have to manage the cache size with atomics etc  - https://github.com/hashicorp/golang-lru

Reply:
done